src/api.js

```diff
@@ -0,0 +1,86 @@
+let debug = require('debug')('app:main');
+let APIBuilder = require('taskcluster-lib-api');
+let taskcluster = require('taskcluster-client');
+let uuid = require('uuid');
+
+let builder = new APIBuilder({
+  title:    'AMQP Messages API Documentation',
+  description: [
+    'This service is responsible for making pulse messages accessible',
+    'from browsers and cli. There are API endpoints to',
+    'bind / unbind to an exchange and pause / resume listening from a queue',
+  ].join('\n'),
+  serviceName: 'tcevents',
+  version: 'v1',
+  errorCodes: {
+    InputValidation:  400, // For JSON schema errors (bad exchange).
+  },
+  context: ['connection'],
+});
+
+builder.declare({
+  method: 'get',
+  route: '/connect/',
+  name: 'connect',
+  description: 'Connect to receive messages',
+  stability: APIBuilder.stability.experimental,
+  // Add input validation yml
+  title: 'Events-Api',
+}, async function(req, res) {
+  console.log("hello");
+  const sendEvent = (kind, data) => {
+    res.write('event: ' + kind + '\n');
+    res.write('data: ' + JSON.stringify(data) + '\n');
+    res.write('\n');
```

Comment:
Maybe merge all the strings before calling `res.write`

Reply:
You should also handle any errors thrown while writing the event. Such errors should trigger you to do cleanup...

You could do a trick like:
```js
let abort; // function to be called to reject the "aborted" promise
const aborted = new Promise((resolve, reject) => abort = reject);

const sendEvent = (kind, data) => {
  try {
     res.write(...);
  } catch(err) {
    abort(err); // abort listening for events
  }
};

try {
   ...
   // create listener
   ...
  
  // wait for the first error, notice that Promise.all returns after the first reject promise.
  await Promise.all([
    aborted,  // something went wrong sending a message
    new Promise((resolve, reject) => listener.once('error', reject)), // listener had an error
    new Promise((resolve, reject) => res.once('finished', reject)),  // request was finished, closed remotely
  ]);
} catch(err) {
  ...
} finally {
}
```